lib/mocha/mockery.rb

```diff
@@ -194,6 +153,13 @@ def logger
 
     private
 
+    def check(action, description, object, method)
+      return if Mocha::Configuration.allow?(action) || (block_given? && !yield)
+      stubbing_error = "stubbing #{description}: #{object.mocha_inspect}.#{method}"
+      raise StubbingError.new(stubbing_error, caller) if Mocha::Configuration.prevent?(action)
+      logger.warn(stubbing_error) if Mocha::Configuration.warn_when?(action)
+    end
+
```

Comment:
I don't understand what this bit of the commit note means:

> I had to create method_exists? temporarily to please rubocop even though it stood out as different from all the other object.foo conditions.

As far as I can see this commit does not introduce `method_exists? `. So I don't understand what you mean by "I had to create method_exists? temporarily". Can you explain?

Reply:
Resolving - the whole branch has been re-written.